### PR TITLE
Added the dunder str to repr.py and updated the file name

### PR DESCRIPTION
**Pull request by isFakeAccount**
- Added the __str_() in the repr.py
- Updated the repr.py filename to str_and_repr.py
- Also added the .idea in gitirnore for jetbrain Ides

### CODE REVIEW

**Review by trag1c — COMMENTED** [review]
I think including `!a` would be nice too :)

**Comment by trag1c** [comment]
It should be made clear that `!s` is the default specifier and that `f"{x!s}"` is equivalent to just `f"{x}"` :+1:

**Comment by isFakeAccount** [comment]
crap, I forgot that it would add the fixed point file here

**Comment by trag1c** [comment]
> crap, I forgot that it would add the fixed point file here

that's why you should use branches 💀💀💀

**Comment by isFakeAccount** [comment]
Gone, Reduced to Atoms

**Review by trag1c — APPROVED** [review]

**Comment by isFakeAccount** [comment]
wtf

**Comment by isFakeAccount** [comment]
#29 Linking issue to this PR

File: src/middle-earth/str_and_repr.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
+class MyCoolClass:
+    def __init__(self, *, value: str) -> None:
+        self.value = value
+
+    def __repr__(self) -> str:
+        """Repr prints the python code to create this python object."""
```
Comment (CircuitSacul) [inline src/middle-earth/str_and_repr.py:6]: I wouldn't state this as fact. Even the stdlib doesn't follow this. Also it technically doesn't "print" anything.

Instead, I'd say something like "Generally, `__repr__` returns the code for creating the instance as a string".

**Review by CircuitSacul — CHANGES_REQUESTED** [review]

File: src/middle-earth/str_and_repr.py (hunk 1 above)
Comment (trag1c) [reply src/middle-earth/str_and_repr.py:6]: This! Alternatively "`__repr__` should return [...]".

File: src/middle-earth/str_and_repr.py [hunk 2]
```diff
@@ -3,7 +3,7 @@ def __init__(self, *, value: str) -> None:
         self.value = value
 
     def __repr__(self) -> str:
-        """Repr prints the python code to create this python object."""
+        """repr returns string containing a printable representation of an object"""
```
Comment (CircuitSacul) [inline src/middle-earth/str_and_repr.py:6]: strings are inherently printable?

maybe "Returns a string containing the code for creating the object"?

Comment (isFakeAccount) [reply src/middle-earth/str_and_repr.py:6]: > strings are inherently printable?
> 
> maybe "Returns a string containing the code for creating the object"?

Yes, that's what the official docs say. https://docs.python.org/3/library/functions.html#repr

Comment (CircuitSacul) [reply src/middle-earth/str_and_repr.py:6]: I still feel like it's a poor description

Comment (isFakeAccount) [reply src/middle-earth/str_and_repr.py:6]: I disagree. Weird thing to nitpick on. Besides is it guaranteed that repr will always yield that object. 

> For many types, this function makes an attempt to return a string that would yield an object with the same value when passed to [eval()](https://docs.python.org/3/library/functions.html#eval)

The official docs do not say *for all types*.

**Comment by isFakeAccount** [comment]
I rebased off the main branch which is why it looks like this

File: .gitignore [hunk 3]
```diff
@@ -157,4 +157,4 @@ cython_debug/
 #  be found at https://github.com/github/gitignore/blob/main/Global/JetBrains.gitignore
 #  and can be added to the global gitignore or merged into this file.  For a more nuclear
 #  option (not recommended) you can uncomment the following to ignore the entire idea folder.
-#.idea/
\ No newline at end of file
+.idea/
```
Comment (MithicSpirit) [inline .gitignore:160]: No EOL at EOF

File: src/middle-earth/str_and_repr.py (hunk 2 above)
Comment (MithicSpirit) [reply src/middle-earth/str_and_repr.py:6]: Even though it's not an absolute requirement I think that it should be indicated on the docstring.

**Review by MithicSpirit — CHANGES_REQUESTED** [review]

File: src/middle-earth/str_and_repr.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        """
+        return f"MyCoolClass(value={self.value!r})"
+
+    def __str__(self) -> str:
+        """str returns the human-readable string for this class"""
+        return f"My Cool Class Value is: {self.value}"
+
+
+if __name__ == "__main__":
+    my_class = MyCoolClass(value="© OpenAI ®")
```
Comment (MithicSpirit) [inline src/middle-earth/str_and_repr.py:24]: I'm not too fond of this example string. Maybe something like `∫ f(x) dx` would be nice as a nod to [James's personal website](https://intfxdx.com/).

File: src/middle-earth/str_and_repr.py [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    print("Printing the output from __repr__()")
+    class_repr = repr(my_class)
+    print("The repr is:", class_repr)
+
+    # Evaluating the repr gives us identical object most of the time.
+    # Except in some occasions like:
+    #
+    # digit_regex = re.compile(r"\d")
+    # x = lambda y: y
+    # o = object()
```
Comment (MithicSpirit) [inline src/middle-earth/str_and_repr.py:35]: Why isn't the repr of these examples printed out?

File: .gitignore [hunk 6]
```diff
@@ -157,4 +157,5 @@ cython_debug/
...
+
```
Comment (MithicSpirit) [inline .gitignore:161]: Now there's an extraneous newline. Not really a problem though.

**Review by MithicSpirit — CHANGES_REQUESTED** [review]

File: src/middle-earth/str_and_repr.py [hunk 7]
```diff
@@ -0,0 +1,32 @@
+import re
+
+
+class MyCoolClass:
+    def __init__(self, *, value: str) -> None:
```
Comment (qexat) [inline src/middle-earth/str_and_repr.py:5]:
```suggestion
    def __init__(self, value: str) -> None:
```
Is making the argument keyword-only really necessary?

**Review by PiCake314 — APPROVED** [review]
Good example. Easy to follow and makes sense.

**Review by Enderchief — APPROVED** [review]
LGTM

**Pull request merged**